cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/runtimejob/DataprocRuntimeJobManager.java

```diff
@@ -359,6 +365,27 @@ private LocalFile getLauncherJar(LocationFactory locationFactory) throws IOExcep
     return DataprocJarUtil.getLauncherJar(locationFactory);
   }
 
+  /**
+   * Check whether delete lifecycle with days since customer time has been enabled on the bucket or not.
+   * @param bucketName
+   * @return true if delete lifecycle with days since custom time is set on the bucket. 
+   */
+  private boolean isDeleteLifecycleEnabled(String bucketName) {
+    Storage storage = getStorageClient();
+    Bucket bucket = storage.get(bucketName);
+    for (BucketInfo.LifecycleRule rule : bucket.getLifecycleRules()) {
+      if (rule.getAction() == null || rule.getCondition() == null ||
+        rule.getCondition().getDaysSinceCustomTime() == null) {
+        continue;
```

Comment:
nit: Does it makes sense to add a log here something like - `Delete lifecycle is disabled on the GCS bucket, artifacts will not be cached`??

Reply:
We do have WARN log on line 193 if either DISABLE_GCS_CACHING is set to true (which is a hidden config and we don't expect customers to use it) or isDeleteLifecycleEnabled returns false.

Reply:
yeah saw that but in case both the conditions are true we might not know the cause of cache disabling. If you think this situation can never occur, can ignore this comment.